DogTales/dog/player.cpp

```diff
@@ -1,21 +1,34 @@
 #include <dog/player.hpp>
 
 namespace dog {
-Player::Player(glm::vec2 const world_space) : m_world_space(world_space) { m_sprite.set_size(size_v); }
+Player::Player(bave::App& app, glm::vec2 const world_space) : m_app(app), m_world_space(world_space) {
+	m_sprite.set_size(size_v);
+}
 
 void Player::tick(bave::Seconds const dt) {
 
 	m_physics.tick(dt);
-	m_sprite.transform.position = m_physics.position;
 
+	auto const& key_state = m_app.get_key_state();
+	auto direction = glm::vec2{};
+	if (key_state.is_pressed(bave::Key::eW) || key_state.is_pressed(bave::Key::eUp)) { direction.y += 1.0f; }
+	if (key_state.is_pressed(bave::Key::eS) || key_state.is_pressed(bave::Key::eDown)) { direction.y -= 1.0f; }
+	if (key_state.is_pressed(bave::Key::eA) || key_state.is_pressed(bave::Key::eLeft)) { direction.x -= 1.0f; }
+	if (key_state.is_pressed(bave::Key::eD) || key_state.is_pressed(bave::Key::eRight)) { direction.x += 1.0f; }
+
+	if (direction.x != 0.0f || direction.y != 0.0f) {
+		direction = glm::normalize(direction);
+		auto const displacement = direction * speed_v * dt.count();
+		m_physics.position += displacement;
```

Comment:
shouldn't movement adjust the player's `acceleration` rather than editing the position directly? and if you don't want movement to have the same "feel" as gravity, you just need custom x-direction multipliers.